### PR TITLE
PRO-3799: Code Cleanup

### PR DESCRIPTION
**Pull request by LucaDelBuonoHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRI-23
(previously https://tools.hmcts.net/jira/browse/PRO-3799)

### Change description ###
This removes the screening questions feature toggle

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: app/config.js [hunk 1]
```diff
@@ -145,7 +144,7 @@ const config = {
     whitelistedPagesAfterPayment: ['/tasklist', '/payment-status', '/documents', '/thankyou', '/check-answers-pdf', '/declaration-pdf', '/sign-out'],
     whitelistedPagesAfterDeclaration: ['/tasklist', '/executors-invites-sent', '/copies-uk', '/assets-overseas', '/copies-overseas', '/copies-summary', '/payment-breakdown', '/payment-breakdown?status=failure', '/payment-status', '/documents', '/thankyou', '/check-answers-pdf', '/declaration-pdf', '/sign-out'],
     hardStopParams: ['will.left', 'will.original', 'iht.completed', 'applicant.executor'],
-    nonIdamPages: ['stop-page/*', 'error', 'sign-in', 'pin-resend', 'pin-sent', 'co-applicant-*', 'pin', 'inviteIdList', 'start-eligibility', 'start-apply', 'new-start-eligibility', 'new-will-left', 'new-will-original', 'new-death-certificate', 'new-deceased-domicile', 'new-applicant-executor', 'new-mental-capacity', 'new-iht-completed', 'new-start-apply'],
+    nonIdamPages: ['error', 'sign-in', 'pin-resend', 'pin-sent', 'co-applicant-*', 'pin', 'inviteIdList', 'start-eligibility', 'will-left', 'will-original', 'death-certificate', 'deceased-domicile', 'applicant-executor', 'mental-capacity', 'iht-completed', 'start-apply'],
```
[the remark this answers is not in the document]
Comment (LucaDelBuonoHMCTS) [reply app/config.js]: Uhm, not sure about this, keeping this in line with all the merges and fixing conflicts has been a bit of a nightmare, like all those rogue handlePost(), persistFormData() and setEligibilityCookie() I just removed. Let me test this in the preview environment and will get back to you.

[the remark this answers is not in the document]
Comment (LucaDelBuonoHMCTS) [reply app/config.js]: Yes, stop pages still need to be whitelisted.

File: app/steps/ui/applicant/executor/index.js [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
+class ApplicantExecutor extends EligibilityValidationStep {
 
     static getUrl() {
-        return '/applicant-executor';
+        return pageUrl;
     }
 
-    nextStepUrl(req, ctx) {
-        return this.next(req, ctx).constructor.getUrl('notExecutor');
+    getContextData(req, res) {
```
Comment (shepda) [inline app/steps/ui/applicant/executor/index.js]: Please add a unit test for `getContextData()`

File: app/steps/ui/deceased/deathcertificate/index.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
 
-class DeathCertificate extends ValidationStep {
+class DeathCertificate extends EligibilityValidationStep {
 
     static getUrl() {
-        return '/death-certificate';
+        return pageUrl;
+    }
+
+    getContextData(req, res) {
```
Comment (shepda) [inline app/steps/ui/deceased/deathcertificate/index.js]: Please add a unit test for `getContextData()`

File: app/steps/ui/deceased/domicile/index.js [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
 
     static getUrl() {
-        return '/deceased-domicile';
+        return pageUrl;
     }
 
-    isComplete(ctx, formdata, featureToggles) {
-        const isEnabled = FeatureToggle.isEnabled(featureToggles, 'screening_questions');
-        return [isEnabled ? true : this.validate(ctx, formdata)[0], 'inProgress'];
+    getContextData(req, res) {
```
Comment (shepda) [inline app/steps/ui/deceased/domicile/index.js]: Please add a unit test for `getContextData()`

File: app/steps/ui/executors/mentalcapacity/index.js [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
 
-class MentalCapacity extends ValidationStep {
+class MentalCapacity extends EligibilityValidationStep {
 
     static getUrl() {
-        return '/mental-capacity';
+        return pageUrl;
+    }
+
+    getContextData(req, res) {
```
Comment (shepda) [inline app/steps/ui/executors/mentalcapacity/index.js]: Please add a unit test for `getContextData()`

File: app/steps/ui/iht/completed/index.js [hunk 6]
```diff
@@ -1,29 +1,30 @@
...
 
-class IhtCompleted extends ValidationStep {
+class IhtCompleted extends EligibilityValidationStep {
 
     static getUrl() {
-        return '/iht-completed';
+        return pageUrl;
+    }
+
+    getContextData(req, res) {
```
Comment (shepda) [inline app/steps/ui/iht/completed/index.js]: Please add a unit test for `getContextData()`

**Pull request merged**